client/src/main/scala/io/finch/client/DecodeResource.scala

```diff
@@ -0,0 +1,28 @@
+package io.finch
+package client
+
+import argonaut._, Argonaut._
+import com.twitter.io.Buf
+import com.twitter.io.Buf.Utf8
+
+trait DecodeResource[A] {
+  def apply(buf: Buf): A 
+}
+
+object DecodeResource {
+
+  implicit def decodeStringResource: DecodeResource[String] = new DecodeResource[String] {
+    def apply(buf: Buf): String = asString(buf)
+  }
+  
+  implicit def decodeJsonResource: DecodeResource[Json] = new DecodeResource[Json] {
+    def apply(buf: Buf): Json = asString(buf)
```

Comment:
I think I'd pretty much always prefer an explicit failure (with information about the parse error) to an empty object in the case of bad input.